honeybee_energy/run.py

```diff
@@ -69,34 +134,63 @@ def run_osw_windows(osw_json, measures_only=True):
-    # check the openstudio directory
-    if not folders.openstudio_path:
-        raise OSError('No OpenStudio installation was found on this machine.\n'
-                      'Install OpenStudio to run energy simulations.')
-
-    # check the input files
-    assert os.path.isfile(osw_json), 'No OSW file found at {}.'.format(osw_json)
-    directory = os.path.split(osw_json)[0]
+    # check the input file
+    directory = _prepare_osw(osw_json)
 
-    # Write the batch file to apply the measures.
+    # Write the batch file to call OpenStudio CLI
     working_drive = directory[:2]
     measure_str = '-m ' if measures_only else ''
     batch = '{}\ncd {}\n"openstudio.exe" run {}-w {}'.format(
         working_drive, folders.openstudio_path, measure_str, osw_json)
-    write_to_file_by_name(directory, 'run_workflow.bat', batch, True)
+    batch_file = os.path.join(directory, 'run_workflow.bat')
+    write_to_file(batch_file, batch, True)
     
     # run the batch file
-    os.system(os.path.join(directory, 'run_workflow.bat'))
+    os.system(batch_file)
     
-    # return the paths to the OSM and IDF
-    osm_file = os.path.join(directory, 'run', 'in.osm')
-    idf_file = os.path.join(directory, 'run', 'in.idf')
+    # output the simulation files
+    return _output_openstudio_files(directory)
 
-    # check that the OSM and IDF files exist
-    osm = osm_file if os.path.isfile(osm_file) else None
-    idf = idf_file if os.path.isfile(idf_file) else None
 
-    return osm, idf
+def run_osw_unix(osw_json, measures_only=True):
+    """Run a .osw file using the OpenStudio CLI on a Unix-based operating system.
+
+    This includes both Mac OS and Linux since a shell will be used to run
+    the simulation.
+    
+    Args:
+        osw_json: File path to a OSW file to be run using OpenStudio CLI.
+        measures_only: Boolean to note whether only the measures should be applied
+            in the runnning of the OSW (True) or the resulting model shoudl be run
+            through EnergyPlus after the measures are aplied to it (False).
+            Default: True.
+    
+    Returns:
+        The following files output from the CLI run.
+        osm -- Path to a .osm file containing all simulation results.
+            Will be None if no file exists.
+        idf -- Path to a .idf file containing properties of the model, including
+            the size of HVAC objects. Will be None if no file exists.
+    """
+    # check the input file
+    directory = _prepare_osw(osw_json)
+
+    # Write the shell script to call OpenStudio CLI
+    measure_str = '-m ' if measures_only else ''
+    shell = '#!/usr/bin/env bash\n\ncd {}\n"openstudio" run {}-w {}'.format(
+        folders.openstudio_path, measure_str, osw_json)
+    shell_file = os.path.join(directory, 'run_workflow.sh')
+    write_to_file(shell_file, shell, True)
+
+    # make the shell script executable using subprocess.check_call
+    # this is more reliable than native Python chmod on Mac
```

Comment:
I don't know what you mean here by native Python chmod but the code below works just fine on Mac:

```python
def change_permission(exec_file):
    st = os.stat(exec_file)
    os.chmod(exec_file, st.st_mode | 0o111)
```

Reply:
I was trying to get your suggestion to work on Rosi's Mac. I tried several variations of it but it just was not working when it was used from Rhino 6's IronPython. Maybe it will work with cPython but I am positive that what I have in place now works from Rhino 6's IronPython on Mac.